python/thunder/utils/datasets.py

```diff
@@ -63,6 +63,44 @@ def generate(self, k=3, npartitions=10, nrows=100, ncols=10, seed=None):
         else:
             return data
 
+class FAData(DataSets):
```

Comment:
I find `FA` slightly ambiguous, and I'm not sure it's quite well known enough by it's abbreviation, compared to say PCA and ICA. I might change the class name to `FactorAnalysisData` and then use `factor` as the shorthand.